platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformStateAccessorSingleton.java

```diff
@@ -0,0 +1,444 @@
+        updatePlatformState(
+                (p, b) -> b.firstVersionInBirthRoundMode(firstVersionInBirthRoundMode.getPbjSemanticVersion()));
+    }
+
+    /**
+     * Get the last round before the birth round mode was enabled, o r -1 if birth round mode has not yet been enabled.
+     *
+     * @return the last round before the birth round mode was enabled
+     */
+    public long getLastRoundBeforeBirthRoundMode() {
+        return getPlatformState().lastRoundBeforeBirthRoundMode();
+    }
+
+    /**
+     * Set the last round before the birth round mode was enabled.
+     *
+     * @param lastRoundBeforeBirthRoundMode the last round before the birth round mode was enabled
+     */
+    public void setLastRoundBeforeBirthRoundMode(final long lastRoundBeforeBirthRoundMode) {
+        updatePlatformState((p, b) -> b.lastRoundBeforeBirthRoundMode(lastRoundBeforeBirthRoundMode));
+    }
+
+    /**
+     * Get the lowest judge generation before the birth round mode was enabled, or -1 if birth round mode has not yet
+     * been enabled.
+     *
+     * @return the lowest judge generation before the birth round mode was enabled
+     */
+    public long getLowestJudgeGenerationBeforeBirthRoundMode() {
+        return getPlatformState().lowestJudgeGenerationBeforeBirthRoundMode();
+    }
+
+    /**
+     * Set the lowest judge generation before the birth round mode was enabled.
+     *
+     * @param lowestJudgeGenerationBeforeBirthRoundMode the lowest judge generation before the birth round mode was
+     *                                                  enabled
+     */
+    public void setLowestJudgeGenerationBeforeBirthRoundMode(final long lowestJudgeGenerationBeforeBirthRoundMode) {
+        updatePlatformState(
+                (p, b) -> b.lowestJudgeGenerationBeforeBirthRoundMode(lowestJudgeGenerationBeforeBirthRoundMode));
+    }
+
+    void migrateToPlatformStateAccessorSingleton() {
+        MerkleNode child = root.getChild(0);
+        // Handling special migration case.
+        // This check will be true in  case of loaded state of version 0.53.
+        if (child instanceof com.swirlds.platform.state.PlatformState platformState) {
+            root.setChild(0, platformSingletonSupplier.get());
+            updatePlatformState(platformState);
+        }
+    }
+
+    /**
+     * Updates values of existing platform state with the values from the source.
+     * @param source source of the new values
+     */
+    void updatePlatformState(PlatformStateAccessor source) {
+        updatePlatformState((p, b) -> {
+            if (source == this) {
+                return ((PlatformStateAccessorSingleton) source)
+                        .getPlatformState()
+                        .copyBuilder();
+            } else {
+                return convert(source).copyBuilder();
+            }
+        });
+    }
+
+    private void updatePlatformState(BiFunction<PlatformState, PlatformState.Builder, PlatformState.Builder> updater) {
```

Comment:
The values in `PlatformState` are always set as a group, like this.

```
        platformState.setRound(round.getRoundNum());
        platformState.setConsensusTimestamp(round.getConsensusTimestamp());
        platformState.setCreationSoftwareVersion(softwareVersion);
        platformState.setRoundsNonAncient(roundsNonAncient);
        platformState.setSnapshot(round.getSnapshot());
```

It seems odd to be making all these copies and doing work every time a value is set, rather than just creating a single `update` method, that accepts all the `PlatformState` values as arguments

Reply:
would you consider current implementation reasonable?

Reply:
I added `PlatformStateAccessor#bulkUpdate` to address this issue.